pkg/cmd/server/admin/create_bootstrap_project_template.go

```diff
@@ -30,7 +34,11 @@ func NewCommandCreateBootstrapProjectTemplate(f *clientcmd.Factory, commandName
 				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
 			}
 
-			template, err := options.CreateBootstrapProjectTemplate()
+			// We can't depend on len(options.NodeSelector) > 0 as node-selector="" is valid
+			// and we want to populate node selector as project annotation only if explicitly set by user
+			checkNodeSelector := cmd.Flag("node-selector").Changed
```

Comment:
> reason for adding this option to project template: user don't have to worry about the key used for node selector, in our case 'openshift.io/node-selector'.

I'm inclined to force them to be aware of it.  They're trying to do an uncommon and complicated thing.  This command isn't intended to give you a template customized for you.  It's way to figure out what template we're using to create your projects by default and use it as a starting point for whatever crazy thing you want.